prisma/schema.sql

```diff
@@ -0,0 +1,158 @@
+  is_archived BOOLEAN DEFAULT false NOT NULL
+);
+
+CREATE TABLE parent (
+  parent_id SERIAL PRIMARY KEY NOT NULL,
+  first_name VARCHAR(255) NOT NULL,
+  last_name VARCHAR(255) NOT NULL,
+  phone_number VARCHAR(50) NOT NULL,
+  email VARCHAR(255) UNIQUE NOT NULL,
+  is_low_income BOOLEAN DEFAULT false,
+  address_line1 VARCHAR(255) NOT NULL,
+  address_line2 VARCHAR(255),
+  postal_code VARCHAR(10) NOT NULL,
+  city_name VARCHAR(50) NOT NULL,
+  province provinces NOT NULL,
+  preferred_language VARCHAR(20) NOT NULL
+);
+
+CREATE TABLE volunteer (
+  volunteer_id SERIAL PRIMARY KEY NOT NULL,
+  first_name VARCHAR(255)  NOT NULL,
+  last_name VARCHAR(255)  NOT NULL,
+  email VARCHAR(255) UNIQUE,
+  phone_number VARCHAR(50),
+  is_valid BOOLEAN DEFAULT false,
+  background_form_link TEXT,
+
+  address_line1 VARCHAR(255),
+  address_line2 VARCHAR(255),
+  postal_code VARCHAR(10),
+  city_name VARCHAR(255),
+  province provinces,
+  
+  preferred_language VARCHAR(20) -- how long is a language name...
+);
+
+CREATE TABLE volunteer_reg (
+  volunteer_id INTEGER NOT NULL,
+  program_id INTEGER NOT NULL,
+  FOREIGN KEY(volunteer_id) REFERENCES volunteer(volunteer_id),
+  FOREIGN KEY(program_id) REFERENCES program(program_id),
+  is_valid BOOLEAN DEFAULT false,
+  PRIMARY KEY (volunteer_id, program_id)
+);
+
+CREATE TABLE coupon_users (
+  program_id INTEGER,
+  parent_id INTEGER,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (program_id) REFERENCES program(program_id) ON DELETE CASCADE,
+  coupon_id VARCHAR(255),
+  PRIMARY KEY (parent_id, program_id)
+);
+
+CREATE TABLE  program_waitlist(
+  program_id INTEGER,
+  parent_id INTEGER,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (program_id) REFERENCES program(program_id) ON DELETE CASCADE,
+  PRIMARY KEY (parent_id, program_id)
+);
+
+CREATE TABLE student(
+  student_id SERIAL PRIMARY KEY NOT NULL,
+  first_name VARCHAR(255)  NOT NULL,
+  last_name VARCHAR(255)  NOT NULL,
+  allergies VARCHAR(255) NOT NULL,
+  additional_info VARCHAR(255) 
+);
+
+CREATE TABLE parent_of_student(
+  parent_id INTEGER,
+  student_id INTEGER,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (student_id) REFERENCES student(student_id) ON DELETE CASCADE,
+  PRIMARY KEY (student_id, parent_id)
+);
+
+CREATE TABLE parent_reg (
+  volunteer_id INTEGER NOT NULL,
+  parent_id INTEGER NOT NULL,
+  student_id INTEGER NOT NULL,
+  program_id INTEGER NOT NULL,
+  is_valid BOOLEAN NOT NULL,
+  FOREIGN KEY (volunteer_id) REFERENCES volunteer(volunteer_id) ON DELETE CASCADE,
+  FOREIGN KEY (parent_id) REFERENCES parent(parent_id) ON DELETE CASCADE,
+  FOREIGN KEY (student_id) REFERENCES student(student_id) ON DELETE CASCADE,
+  FOREIGN KEY (program_id) REFERENCES program(program_id) ON DELETE CASCADE,
+  PRIMARY KEY (student_id, program_id)
+);
+
+CREATE TABLE "admin" (
```

Comment:
Why does admin have quotations around it?

Reply:
its some kind of keyword, like the IDE highlighted it the same color as other keywords
victor added it he might know more